libraries/mapping/engines/cesium/src/lib/CustomViewer/CustomViewer.tsx

```diff
@@ -124,6 +125,10 @@ function CustomViewer(props: CustomViewerProps) {
 
 
   const [viewer, setComponentStateViewer] = useState<Viewer | null>(null);
+  const previousViewerRef = useRef<Viewer | null>(null); // track viewer changes
```

Comment:
@d4v3000  does not seem the best way to have useEffect tracking changes, maybe custom hooks with own refs declutter this?